libsolidity/interface/CompilerStack.cpp

```diff
@@ -656,8 +675,40 @@ bool CompilerStack::isRequestedContract(ContractDefinition const& _contract) con
 	return false;
 }
 
+bool CompilerStack::assemble()
```

Comment:
What do you mean exactly with Solidity compilation? That this should be handled in `CompilerStack::compile()`? Without having such method?

Reply:
I meant that assembling is a part of the normal compilation path from Solidity source to bytecode:
https://github.com/ethereum/solidity/blob/e769c7981391bc647cecc0b328c314a5c4f36e75/libsolidity/interface/CompilerStack.cpp#L1354

This part is not using the new assembly stack. The `assembly()` called here is the other overload, that does not use it and seems completely independent from yours.

Reply:
Still not done. I understand what you're meaning here, but I need to think about the structure again.

Reply:
I just went through the review comments again. Right now I just introduced a method `CompilerStack::importFromEVMAssemblyStack(evmasm::EVMAssemblyStack const& _stack)` - it will be used to setup `m_contracts` and fills the corresponding `evmAssembly`, `evmRuntimeAssembly`, `object` and `runtimeObject` contract items. It will also setup `m_sources`. From my point of view there is no real need to put that in the current compilation pipeline. This is just used in `CommandLineInterface::assembleFromEvmAssemblyJson()`. The main thing why I use the `CompilerStack` here is, that `CompilerStack::importFromEVMAssemblyStack(..)` will setup the stack, so that `CommandLineInterface::handleCombinedJSON()`,`CommandLineInterface::handleBytecode(..)` and `CommandLineInterface::handleEVMAssembly(..)` can just be reused here, without introducing any additional logic for the import case.